### ISSUE TITLE
Install request: h5utils

### ISSUE THREAD

**Issue by heatherkellyucl** [UCL-RITS/rcps-buildscripts#58]
IN01881718

http://ab-initio.mit.edu/wiki/index.php/H5utils

**Comment by heatherkellyucl** [comment]
If you try to build this with libmatheval (optional to enable h5math) then configure gets an error and configures without it.

```
configure:4577: checking for evaluator_get_variables in -lmatheval
configure:4612: gcc -o conftest -g -O2   conftest.c -lmatheval  -lz -lm  >&5
/shared/ucl/apps/libmatheval/1.1.11/gnu-4.9.2/lib/libmatheval.so: undefined reference to `yywrap'
collect2: error: ld returned 1 exit status

configure:4640: result: no
configure:4653: WARNING: can't find libmatheval: won't be able to compile h5math
```

yywrap is part of flex: http://flex.sourceforge.net/manual/I-get-an-error-about-undefined-yywrap_0028_0029_002e.html#I-get-an-error-about-undefined-yywrap_0028_0029_002e
> **I get an error about undefined yywrap().**

> You must supply a `yywrap()` function of your own, or link to `libfl.a` (which provides one), or use
>    `%option noyywrap`
> in your source to say you don't want a `yywrap()` function. 

Linking to the shared library does not work as it can't find yylex, which is apparently a bug with the shared library for flex 2.5.39:
```
./configure --prefix=/dev/shm/h5utils LIBS="-lhdf5 -lz -lm -lfl"

configure:2791: gcc    conftest.c -lhdf5 -lz -lm -lfl >&5
/shared/ucl/apps/flex/2.5.39/gnu-4.9.2/lib/libfl.so: undefined reference to `yylex'
```

Linking to the static libfl.a does work, then you get a libpng error:
```
./configure --prefix=/dev/shm/h5utils LIBS="-lhdf5 -lz -lm -lfl"
configure:2791: gcc    conftest.c -lhdf5 -lz -lm -lfl >&5
/shared/ucl/apps/flex/2.5.39/gnu-4.9.2/lib/libfl.so: undefined reference to `yylex'

gcc  -g -O2   -o h5math h5math.o arrayh5.o h5utils.o -lmatheval -lhdf5 -lz -lm -lhdf5 -lz -lm /shared/ucl/apps/flex/2.5.39/gnu-4.9.2/lib/libfl.a
gcc -DHAVE_CONFIG_H -I.     -g -O2 -MT h5topng.o -MD -MP -MF .deps/h5topng.Tpo -c -o h5topng.o h5topng.c
mv -f .deps/h5topng.Tpo .deps/h5topng.Po
gcc -DHAVE_CONFIG_H -I.     -g -O2 -MT writepng.o -MD -MP -MF .deps/writepng.Tpo -c -o writepng.o writepng.c
In file included from /usr/include/png.h:452:0,
                 from writepng.c:28:
writepng.c: In function ‘writepng’:
writepng.c:312:24: error: dereferencing pointer to incomplete type
      if (setjmp(png_ptr->jmpbuf)) {
                        ^
writepng.c:437:19: error: dereferencing pointer to incomplete type
      free(info_ptr->palette);
                   ^
make[1]: *** [writepng.o] Error 1
```

I knew h5utils sounded familiar - it is often used with meep to convert output and a while ago a user had this same problem when installing their own copy. There is a patch for h5utils (libpng changed at version 15).

https://github.com/hmaarrfk/meep-rpm/blob/master/h5utils-1.12.1-png_get_.patch

**Comment by heatherkellyucl** [comment]
Available on Legion as `h5utils/1.12.1`.

- [ ] Install on Grace when upgrade finished.
